codersdk/workspacebuilds.go

```diff
@@ -19,6 +19,14 @@ const (
 	WorkspaceTransitionDelete WorkspaceTransition = "delete"
 )
 
+type BuildReason string
+
+const (
+	BuildReasonInitiator BuildReason = "initiator"
+	BuildReasonAutostart BuildReason = "autostart"
+	BuildReasonAutostop  BuildReason = "autostop"
+)
```

Comment:
Can we doc when each of these is used and why? Maybe how to use it too? Eg, if `initiator` look at `initiator_id`